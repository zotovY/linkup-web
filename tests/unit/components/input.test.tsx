/// <reference types="@types/jest" />
import { render } from "../../test-utils";
import '@testing-library/jest-dom';
import 'jest-styled-components'

import Input, { Props } from "@/components/input";
import { cleanup } from "@testing-library/react";

describe("Testing input component", () => {

    afterEach(cleanup);

    const props: Props = {
        placeholder: "Hello my friend!",
    }

    /**
     * No need for testing onClick and so one props
     * because Input component is pure styled component
     * I'm sure about the libraries I use and do not
     * consider it necessary to rewrite the tests of
     * the styled command
     */

    test("should renders correctly", () => {
        const { asFragment } = render(<Input { ...props } />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("should renders correctly with error", () => {
        const { asFragment } = render(<Input { ...props } error={true} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
