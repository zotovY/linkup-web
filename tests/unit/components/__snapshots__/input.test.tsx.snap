// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing input component should renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  border: 1.5px solid #D9DDE9;
  padding: 7px 14px;
  border-radius: 10px;
  font-size: 15px;
  line-height: 28px;
  outline: none;
  background: white;
  z-index: 10;
  width: 100%;
  -webkit-transition: border 200ms;
  transition: border 200ms;
}

.c0:focus {
  border: 1.5px solid #14142B;
}

<input
    class="c0"
    placeholder="Hello my friend!"
  />
</DocumentFragment>
`;

exports[`Testing input component should renders correctly with error 1`] = `
<DocumentFragment>
  .c0 {
  border: 1.5px solid #D9DDE9;
  padding: 7px 14px;
  border-radius: 10px;
  font-size: 15px;
  line-height: 28px;
  outline: none;
  background: white;
  z-index: 10;
  width: 100%;
  -webkit-transition: border 200ms;
  transition: border 200ms;
  border: 1.5px solid #ED2E7E !important;
  color: #ED2E7E;
}

.c0:focus {
  border: 1.5px solid #14142B;
}

<input
    class="c0"
    placeholder="Hello my friend!"
  />
</DocumentFragment>
`;
